From c79fa6f95efabee924db78b3b6d90ae494f45119 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Mon, 22 Oct 2018 15:53:41 +0300
Subject: [PATCH 760/767] crypto: caam/qi2 - rate-limit enqueue failure prints
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Avoid console being flooded with prints in case HW is too busy to accept
new enqueue requests.

Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.12.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/caamalg_qi2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/caamalg_qi2.c b/drivers/crypto/caam/caamalg_qi2.c
index 26a1437cfadc..940920c21bde 100644
--- a/drivers/crypto/caam/caamalg_qi2.c
+++ b/drivers/crypto/caam/caamalg_qi2.c
@@ -5796,8 +5796,8 @@ int dpaa2_caam_enqueue(struct device *dev, struct caam_request *req)
 			break;
 	}
 
-	if (unlikely(err < 0)) {
-		dev_err(dev, "Error enqueuing frame: %d\n", err);
+	if (unlikely(err)) {
+		dev_err_ratelimited(dev, "Error enqueuing frame: %d\n", err);
 		goto err_out;
 	}
 
-- 
2.17.0

